spec/testingFramework.js

```diff
@@ -0,0 +1,80 @@
+"use strict";
```

Comment:
A very extensive and thorough testing framework set up, it looks great, what are the reasons for using listCounters and subListCounters?

Reply:
these counters just mean that when you create a new HTML list they have a different ID to the last one made. Not essential but thought it could be useful if I ever want to expand the framework